src/components/RouteCard.tsx

```diff
@@ -1,20 +1,93 @@
-import { Suspense, type VoidComponent } from 'solid-js'
+import { createSignal, createEffect, Suspense, Show, type Component } from 'solid-js'
 import dayjs from 'dayjs'
 
 import Avatar from '~/components/material/Avatar'
-import Card, { CardContent, CardHeader } from '~/components/material/Card'
+import { CardContent, CardHeader } from '~/components/material/Card'
 import Icon from '~/components/material/Icon'
+import RouteOptions from '~/components/RouteOptions'
 import RouteStaticMap from '~/components/RouteStaticMap'
 import RouteStatistics from '~/components/RouteStatistics'
+import Timeline from './Timeline'
 
-import type { RouteSegments } from '~/types'
+import type { Route, RouteSegments } from '~/types'
 
-const RouteHeader = (props: { route: RouteSegments }) => {
-  const startTime = () => dayjs(props.route.segment_start_times[0])
-  const endTime = () => dayjs(props.route.segment_end_times.at(-1))
+import { reverseGeocode } from '~/map'
 
-  const headline = () => startTime().format('ddd, MMM D, YYYY')
-  const subhead = () => `${startTime().format('h:mm A')} to ${endTime().format('h:mm A')}`
+const [showRouteOptionsCard, setShowRouteOptionsCard] = createSignal(false)
+const [currentCardRoute, setCurrentCardRoute] = createSignal<Route | undefined>(undefined)
+
+const RouteOptionsCard: Component = () => {
+  const [isMdOrLarger, setIsMdOrLarger] = createSignal(false)
```

Comment:
You can use tailwind selectors for things like this, instead of javascript
https://tailwindcss.com/docs/responsive-design